cvat/apps/engine/rq_job_handler.py

```diff
@@ -4,80 +4,301 @@
+        jid = job_id(db_obj)
+
+        user = request.user
+
+        return cls(
+            user=UserInfo(
+                id=getattr(user, "id", None),
+                username=getattr(user, "username", None),
+                email=getattr(user, "email", None),
+            ),
+            request=RequestInfo(
+                uuid=request.uuid,
+                timestamp=timezone.localtime(),
+            ),
+            org_id=oid,
+            org_slug=oslug,
+            project_id=pid,
+            task_id=tid,
+            job_id=jid,
+        ).to_dict()
+
+
+@attrs.define(kw_only=True)
+class ExportRQMeta(BaseRQMeta):
+    # will be changed to ExportResultInfo in the next PR
+    result_url: str | None = attrs.field(validator=[optional_str_validator], default=None)
+
+    @staticmethod
+    def _get_resettable_fields() -> list[RQJobMetaField]:
+        base_fields = BaseRQMeta._get_resettable_fields()
+        return base_fields + [RQJobMetaField.RESULT]
+
+    @classmethod
+    def build_for(
+        cls,
+        *,
+        request: ExtendedRequest,
+        db_obj: Model | None,
+        result_url: str | None,
+    ):
+        base_meta = BaseRQMeta.build(request=request, db_obj=db_obj)
+
+        return cls(
+            **base_meta,
+            result_url=result_url,
+        ).to_dict()
+
+
+@attrs.define(kw_only=True)
+class ImportRQMeta(BaseRQMeta):
+    # immutable && optional fields
+    tmp_file: str | None = attrs.field(
+        validator=[optional_str_validator], default=None, on_setattr=attrs.setters.frozen
+    )
+
+    # mutable fields
+    task_progress: float | None = attrs.field(
+        validator=[optional_float_validator], default=None, on_setattr=on_setattr
+    ) # used when importing project dataset
+
+    @staticmethod
+    def _get_resettable_fields() -> list[RQJobMetaField]:
+        base_fields = BaseRQMeta._get_resettable_fields()
+
+        return base_fields + [RQJobMetaField.TASK_PROGRESS]
+
     @classmethod
-    def reset_meta_on_retry(cls, meta_to_update: dict[RQJobMetaField, Any]) -> dict[RQJobMetaField, Any]:
-        resettable_fields = cls.get_resettable_fields()
+    def build_for(
+        cls,
+        *,
+        request: ExtendedRequest,
+        db_obj: Model | None,
+        tmp_file: str | None = None,
+    ):
+        base_meta = BaseRQMeta.build(request=request, db_obj=db_obj)
 
-        return {
-            k: v for k, v in meta_to_update.items() if k not in resettable_fields
-        }
+        return cls(
+            **base_meta,
+            tmp_file=tmp_file,
+        ).to_dict()
 
 class RQJobMetaField:
     # common fields
     FORMATTED_EXCEPTION = "formatted_exception"
-    REQUEST = 'request'
-    USER = 'user'
-    PROJECT_ID = 'project_id'
-    TASK_ID = 'task_id'
-    JOB_ID = 'job_id'
-    ORG_ID = 'org_id'
-    ORG_SLUG = 'org_slug'
-    STATUS = 'status'
-    PROGRESS = 'progress'
-    TASK_PROGRESS = 'task_progress'
+    REQUEST = "request"
```

Comment:
Did you run this file through black? If so, could you delete it from the exceptions in `pyproject.toml`?